x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/retention/ExpiredAnnotationsRemover.java

```diff
@@ -0,0 +1,147 @@
+
+    private final AnomalyDetectionAuditor auditor;
+    private final ThreadPool threadPool;
+
+    public ExpiredAnnotationsRemover(OriginSettingClient client, Iterator<Job> jobIterator, TaskId parentTaskId,
+                                     AnomalyDetectionAuditor auditor, ThreadPool threadPool) {
+        super(client, jobIterator, parentTaskId);
+        this.auditor = Objects.requireNonNull(auditor);
+        this.threadPool = Objects.requireNonNull(threadPool);
+    }
+
+    @Override
+    Long getRetentionDays(Job job) {
+        return job.getAnnotationsRetentionDays();
+    }
+
+    @Override
+    protected void removeDataBefore(
+        Job job,
+        float requestsPerSecond,
+        long latestTimeMs,
+        long cutoffEpochMs,
+        ActionListener<Boolean> listener
+    ) {
+        DeleteByQueryRequest request = createDBQRequest(job, requestsPerSecond, cutoffEpochMs);
+        request.setParentTask(getParentTaskId());
+
+        client.execute(DeleteByQueryAction.INSTANCE, request, new ActionListener<>() {
+            @Override
+            public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+                try {
+                    if (bulkByScrollResponse.getDeleted() > 0) {
+                        auditAnnotationsWereDeleted(job.getId(), cutoffEpochMs);
+                    }
+                    listener.onResponse(true);
+                } catch (Exception e) {
+                    onFailure(e);
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(new ElasticsearchException("Failed to remove expired annotations for job [" + job.getId() + "]", e));
+            }
+        });
+    }
+
+    private static DeleteByQueryRequest createDBQRequest(Job job, float requestsPerSec, long cutoffEpochMs) {
+        QueryBuilder query = QueryBuilders.boolQuery()
+            .filter(QueryBuilders.termQuery(Job.ID.getPreferredName(), job.getId()))
+            .filter(QueryBuilders.rangeQuery(Annotation.TIMESTAMP.getPreferredName()).lt(cutoffEpochMs).format("epoch_millis"))
+            .filter(QueryBuilders.termQuery(Annotation.CREATE_USERNAME.getPreferredName(), XPackUser.NAME));
+        DeleteByQueryRequest request = new DeleteByQueryRequest(AnnotationIndex.READ_ALIAS_NAME)
+            .setSlices(AbstractBulkByScrollRequest.AUTO_SLICES)
+            .setBatchSize(AbstractBulkByScrollRequest.DEFAULT_SCROLL_SIZE)
+            // We are deleting old data, we should simply proceed as a version conflict could mean that another deletion is taking place
+            .setAbortOnVersionConflict(false)
+            .setTimeout(DEFAULT_MAX_DURATION)
+            .setRequestsPerSecond(requestsPerSec)
+            .setQuery(query);
+
+        // _doc is the most efficient sort order and will also disable scoring
+        request.getSearchRequest().source().sort(ElasticsearchMappings.ES_DOC);
```

Comment:
Adding a sort is faster than not adding a sort at all??

EDIT: I see this was added to other queries in this stuff by @droberts195 with the intent of disabling scoring. But, this query only contains filter queries, I think scoring is already disabled if your query only contains filters.

Reply:
I'd also think scoring is out of picture with just `filter` clauses.
I've just removed this "sort" statement.